iceoryx_posh/source/roudi/service_registry.cpp

```diff
@@ -26,16 +26,17 @@ ServiceRegistry::ServiceDescriptionEntry::ServiceDescriptionEntry(const capro::S
 {
 }
 
-cxx::expected<ServiceRegistry::Error> ServiceRegistry::add(const capro::ServiceDescription& serviceDescription) noexcept
+cxx::expected<ServiceRegistry::Error> ServiceRegistry::add(const capro::ServiceDescription& serviceDescription,
+                                                           ReferenceCounter_t ServiceDescriptionEntry::*count)
 {
     auto index = findIndex(serviceDescription);
     if (index != NO_INDEX)
     {
-        // multiple publishers with the same service descripion are possible
+        // multiple entries with the same service descripion are possible
         // and we just increase the count in this case (multi-set semantics)
         // entry exists, increment counter
         auto& entry = m_serviceDescriptions[index];
-        entry->count++;
+        ((*entry).*count)++;
```

Comment:
Oh boy, seeing this in action I'm not sure anymore it was a good idea to give you the hint with the member pointer 😅

Nevertheless, let's keep it

Reply:
There is no easier syntax for it. You may be able to get rid of one pair of braces but then it looks even more arcane.

Reply:
Yeah, why cant we use a normal pointer instead?

Reply:
@elfenpiff the problem is this
```cpp
void doSomethingWithFooOnBar(Foo& foo, pointerToSomeMemberOfBarWhichNeedsToBeFound)
{
    for(auto& bar: allBars) {
        if(bar == foo.someCondition) {
            bar.pointerToSomeMemberOfBarWhichNeedsToBeFound = 42;
        }
    }
}
```